components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -17761,6 +17762,736 @@ public void deleteTenantTheme(int tenantId) throws APIManagementException {
+                    }
+                    String newUUID = addArtifact(connection, themeContent, DevPortalConstants.DRAFTED_ORG_THEME);
+                    updateDraftedArtifactForOrg(connection, organization, newUUID);
+                } else {
+                    String newUUID = addArtifact(connection, themeContent, DevPortalConstants.DRAFTED_ORG_THEME);
+                    insertNewOrgWithDraftedArtifact(connection, organization, newUUID);
+                }
+                connection.commit();
+            } catch (SQLException e) {
+                connection.rollback();
+                handleException("Failed to import drafted organization theme for organization " + organization, e);
+            }
+        } catch (SQLException e) {
+            handleException("Database connection error while importing drafted organization theme for organization " + organization, e);
+        }
+    }
+
+
+    /**
+     * Updates the organization theme status as published or unpublished.
+     *
+     * @param organization Organization name.
+     * @param action       Action to perform ("PUBLISH" or "UNPUBLISH").
+     * @throws APIManagementException If a database error occurs.
+     */
+    public void updateOrgThemeStatusAsPublishedOrUnpublished(String organization, String action) throws APIManagementException {
+        try (Connection connection = APIMgtDBUtil.getConnection()) {
+            connection.setAutoCommit(false);
+            try {
+                if (DevPortalConstants.PUBLISH.equals(action)) {
+                    String draftedArtifact = getDraftedArtifactForOrg(connection, organization);
+                    if (draftedArtifact != null) {
+                        InputStream artifactContent = getArtifactContent(connection, draftedArtifact);
+                        String newUUID = addArtifact(connection, artifactContent, DevPortalConstants.PUBLISHED_ORG_THEME);
+                        updatePublishedArtifactForOrg(connection, organization, newUUID);
+                        removeArtifact(connection, draftedArtifact);
+                    } else {
+                        log.warn("ID cannot be found in drafted state. Aborting the request");
+                    }
+                } else {
+                    String publishedArtifact = getPublishedArtifactForOrg(connection, organization);
+                    if (publishedArtifact != null) {
+                        InputStream artifactContent = getArtifactContent(connection, publishedArtifact);
+                        String newUUID = addArtifact(connection, artifactContent, DevPortalConstants.DRAFTED_ORG_THEME);
+                        updateDraftedArtifactForOrg(connection, organization, newUUID);
+                        removeArtifact(connection, publishedArtifact);
+                    } else {
+                        log.warn("ID cannot be found in published state. Aborting the request");
```

Comment:
We cannot simply abort the request in the DB level, but need to properly populate the error to in REST API level as well.

Reply:
+1, throw errors to REST by [6th set of required changes](https://github.com/wso2/carbon-apimgt/pull/12874/commits/a6d6cd8b552949be6ea2ae144a5bfc530dc4eaf8)